src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/CompiledPageRouteModelProvider.cs

```diff
@@ -58,26 +69,63 @@ private void EnsureCache()
                     rootDirectory = rootDirectory + "/";
                 }
 
+                var areaRootDirectory = _pagesOptions.AreaRootDirectory;
+                if (!areaRootDirectory.EndsWith("/", StringComparison.Ordinal))
+                {
+                    areaRootDirectory = areaRootDirectory + "/";
+                }
+
                 var cachedApplicationModels = new List<PageRouteModel>();
                 foreach (var viewDescriptor in GetViewDescriptors(_applicationManager))
                 {
-                    if (!viewDescriptor.RelativePath.StartsWith(rootDirectory, StringComparison.OrdinalIgnoreCase))
+                    PageRouteModel model = null;
+                    if (viewDescriptor.RelativePath.StartsWith(rootDirectory, StringComparison.OrdinalIgnoreCase))
                     {
-                        continue;
+                        model = GetPageRouteModel(rootDirectory, viewDescriptor);
+                    }
+                    else if (_pagesOptions.EnableAreas && viewDescriptor.RelativePath.StartsWith(areaRootDirectory, StringComparison.OrdinalIgnoreCase))
```

Comment:
Kinda interestingly this means that you can't have areas that differ only by case. I think it's probably fine, but worth bringing up. That makes sense to me because routing is already case-insensitive.

Reply:
It's already problematic with precompiled views since we don't allow two view paths to differ in case.

Reply:
Yeah, I think it makes for a better experience to be case insensitive as having areas/files differ only in casing (and rely on that) is kinda edgy.